### PR TITLE
Add pipeline foundation: types, strategy traits, and contour/join implementations

### PR DESCRIPTION
**Pull request by altendky**
## Summary

- Establish the `mujou-pipeline` type system: `Point`, `Polyline`, `Dimensions`, `PipelineConfig` (with spec-matching defaults), and `PipelineError` (via `thiserror`)
- Define pluggable algorithm strategy patterns using enum dispatch: `ContourTracer` trait + `ContourTracerKind`, `PathJoiner` trait + `PathJoinerKind`
- Implement `BorderFollowing` contour tracer (wraps `imageproc::contours::find_contours`) and `StraightLine` path joiner (concatenates contour points)

## Details

This is the first step of Phase 1 (Core Pipeline) from the [implementation roadmap](docs/src/project/implementation.md). It lays the foundation that all subsequent pipeline modules (`grayscale`, `blur`, `edge`, `simplify`, `optimize`, `mask`) will build on.

### Design decisions

- **Enum dispatch** for strategy selection (not trait objects or generics) — simple, `Clone`/`Default`-friendly, matches the small fixed set of algorithms
- **No serde** on types yet (deferred until actually needed)
- Both `BorderFollowing` and `StraightLine` are **fully functional**, not stubs
- All public functions have doc comments; all methods that can be `const fn` are `const fn`

### New files

| File | Contents |
|------|----------|
| `crates/mujou-pipeline/src/types.rs` | `Point`, `Polyline`, `Dimensions`, `PipelineConfig`, `PipelineError` |
| `crates/mujou-pipeline/src/contour.rs` | `ContourTracer` trait, `ContourTracerKind` enum, `BorderFollowing` impl |
| `crates/mujou-pipeline/src/join.rs` | `PathJoiner` trait, `PathJoinerKind` enum, `StraightLine` impl |
| `crates/mujou-pipeline/src/lib.rs` | Module declarations + re-exports |

### Tests

24 new tests covering:
- Point construction, equality, distance, copy semantics
- Polyline accessors and edge cases (empty)
- `PipelineConfig::default()` matches all values from the pipeline spec
- `PipelineError` display strings
- Contour tracing on synthetic images (empty, single pixel, rectangle)
- Path joining (empty, single, multiple contours, point count preservation)

### CODE REVIEW

File: crates/mujou-pipeline/src/contour.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        // A single white pixel produces a contour with only 1 point,
+        // which our filter removes (need >= 2 points for a polyline).
+        let mut img = GrayImage::new(10, 10);
+        img.put_pixel(5, 5, image::Luma([255]));
+        let result = ContourTracerKind::BorderFollowing.trace(&img);
+        // Single pixel may produce a very short contour; we only keep >= 2 points.
+        for polyline in &result {
+            assert!(polyline.len() >= 2);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (altendky) [reply crates/mujou-pipeline/src/contour.rs]: Good catch on the vacuous assertion. I verified that `imageproc` border-following on a single pixel actually produces a contour with only 1 point here, so it *is* filtered out and `result` is empty. Replaced the for-loop with an explicit `assert!(result.is_empty())` so the test will actually catch any regression.

Fixed in https://github.com/altendky/mujou/commit/55c9fed487adc9669b40b2f896f657169bb0df5d

File: crates/mujou-pipeline/src/types.rs [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    /// Euclidean distance to another point.
+    #[must_use]
+    pub fn distance(self, other: Self) -> f64 {
+        self.distance_squared(other).sqrt()
+    }
+}
+
+/// A sequence of connected points forming a path segment.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Polyline(pub Vec<Point>);
```
[the remark this answers is not in the document]
Comment (altendky) [reply crates/mujou-pipeline/src/types.rs]: Made the inner field private and added `into_points(self) -> Vec<Point>` for callers that need ownership. https://github.com/altendky/mujou/commit/17f7fc2d6a159e534a2072446bcac277e825cc3f
